### PR TITLE
various miscelaneous

### PR DESCRIPTION
**Pull request by bholloway**

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -40,10 +40,10 @@ yargs.getInstance()
     'Tasks include:',
     yargs.listTasks().join(', ')
   ].join('\n')))
-  .example('$0', 'Interactive menu')
-  .example('$0 -v', 'Display the version of angularity')
-  .example('$0 -h <task name>', 'Get help on a particular task')
-  .example('$0 <task name>', 'Run the given task')
+  .example('angularity', 'Interactive menu')
```
Comment (bholloway) [inline bin/cli.js]: The $0 feature of yargs is meant to simplify examples by resolving to the name of the executable but doesn't work properly in Windows.

Expected `angularity` actual `c:/full/path/to/angularity/global/install`.

Instead I have just written `angularity` explicitly.

File: lib/config/defaults.js [hunk 2]
```diff
@@ -143,10 +143,10 @@ function commit() {
  */
 function readFile(filePath, field) {
   if (filePath && fs.existsSync(filePath)) {
-    var data = fs.readFileSync(filePath);
+    var data = fs.readFileSync(filePath).toString();
     try {
       var json = JSON.parse(data)
-      return field ? (json[field] || {}) : field;
+      return field ? (json[field] || {}) : json;
     } catch(error) {
```
Comment (bholloway) [inline lib/config/defaults.js]: Fix bug where the `server` and `watch` tasks did not respect settings in `angular.json`.

File: tasks/init.js [hunk 3]
```diff
@@ -23,6 +23,7 @@ var cliArgs;
 var config = defaults.getInstance('init')
   .file(platform.userHomeDirectory(), '.angularity')
   .defaults({
+    subdir     : true,
```
Comment (bholloway) [inline tasks/init.js]: subdirectory is default for initialisation task

File: tasks/init.js [hunk 4]
```diff
@@ -215,7 +222,7 @@ yargs.getInstance('init')
     default : config.get('gitignore')
   })
   .options('ide', {
-    describe: 'Initialise IDE ' + IDE_LIST.join('|'),
+    describe: 'Run an IDE initialisation task',
```
Comment (bholloway) [inline tasks/init.js]: The original description was misleading when the list contains only one item

File: tasks/webstorm.js [hunk 5]
```diff
@@ -357,6 +357,8 @@ function executablePath() {
  * @returns {string} A true concatentated path where found, else undefined
  */
 function maximisePath() {
+
```
Comment (bholloway) [inline tasks/webstorm.js]: This method is resolving a path that contains some combination of strings and `Regexp`. Previous algorithm performed a directory list for both types of element, even though it was only needed for the `Regexp` element.

There is an obscure bug on windows where a shortcut file (found in `AppData` on some windows machines) is crashing `fs` operations. The new algorithm should not be any worse and should hopefully fix the problem.

Tested on Mac and Windows, but not yet on a machine that exhibits the bug.

File: tasks/init.js [hunk 6]
```diff
@@ -123,23 +124,29 @@ yargs.getInstance('init')
...
+    'Where run on an exising project existing files will not be altered, delete existing files in order to change ' +
+    'properties.',
+    '',
+    'Both the npm and bower packages are initially set private which you will need to clear in order to publish.',
+    '',
+    'Available IDE tasks include:',
+    IDE_LIST.join(', '),
+    '',
+    'Any given IDE is initialised per its task defaults. Use the task separately to review these options.'
   ].join('\n')))
```
Comment (bguiz) [inline tasks/init.js]: - This is beginning to look like we should read the description in from a file - there's quite a lot of string concatentation going on here.
- Do either regex replacements or templating for the dynamic bits
- Question: Do we really need those dynamic bits, or would it be OK to simply hard code "app" here?

File: tasks/webstorm.js (hunk 5 above)
Comment (bguiz) [inline tasks/webstorm.js]: Sweet!

File: tasks/webstorm.js [hunk 7]
```diff
@@ -368,27 +370,31 @@ function maximisePath() {
...
+        return null;
+      } else {
+        elements[i] = matches[0];
+      }
     } else {
-      elements[i] = matches[0];
+      elements[i] = String(elements[i]);
     }
   }
+
```
Comment (bguiz) [inline tasks/webstorm.js]: Perhaps we could do with some one liner comments describing the code flow here, too much guesswork to figure out what is going on here.

File: tasks/init.js (hunk 6 above)
Comment (bholloway) [inline tasks/init.js]: That is a good idea, even the whole thing may be best as `init.json`. Please comment against AF-570 or AF-523, whichever is most appropriate, so we don't loose the idea.

On balance I think I agree about `app`.

**Pull request merged**